api/v1/submissions/SubmissionHandler.inc.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/**
+ * @file api/v1/submission/SubmissionHandler.inc.php
+ *
+ * Copyright (c) 2014-2018 Simon Fraser University
+ * Copyright (c) 2003-2018 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class SubmissionHandler
+ * @ingroup api_v1_submission
+ *
+ * @brief Handle API requests for submission operations.
+ *
+ */
+
+import('lib.pkp.api.v1.submissions.PKPSubmissionHandler');
+import('classes.core.ServicesContainer');
+
+class SubmissionHandler extends PKPSubmissionHandler {
+
+	/**
+	 * Constructor
+	 */
+	public function __construct() {
+		$this->_handlerPath = 'submissions';
+		$roles = array(ROLE_ID_MANAGER, ROLE_ID_SUB_EDITOR, ROLE_ID_ASSISTANT, ROLE_ID_REVIEWER, ROLE_ID_AUTHOR);
+		$this->_endpoints = array(
+			'GET' => array (
```

Comment:
Some of these are identical to those in OJS, right? Is it possible to share more of this in pkp-lib?

Reply:
I have already moved shared handlers' callbacks methods to `PKPSubmissionHandler` class. These are endpoints definitions. Even though some are similar I have left them in subclass for clarity's sake. Do you want me to move common endpoints definitions to the parent class as well?

Reply:
It's nothing crucial, so I'll leave it up to you. Where the handler code is all 100% shared in the parent class, I prefer to have the endpoint definition there too.